### ISSUE TITLE
Error with Swarm LRMs in BTA cause NRE

### ISSUE THREAD

**Issue by IceRaptor** [BattletechModders/CBTBehaviorsEnhanced#29]
From Tstir in BTA #tech-support form:

So, I've run into an issue that it seems @vidarion ran into last Wednesday with Swarm LRMs. Just had the same thing happen to me, in which you use Swarm LRMs to attack ground, the missiles will strike their targets, and then the unit gets stuck. Can't move it, can't end turn, nada. The only thing I could see that might be a cause, is with the Swarm targeting, missiles were impacting mechs that were in the fog of war and not on sensors. At least to my knowledge, if they popped up as a sensor blip, they worked okay, and if the mech was in range of the swarm retargeting off a miss but was in the fog completely obscured, bam! That unit is toast and you effectively need to restart the mission, or withdraw.

Had the same bug last night on a flashpoint I was doing and thought it was a fluke, tried not to use it too much. Since I've had a bit more time today, did a few more missions and played around with attack ground and seemed to work fine. However, when I just tried it as of a few minutes ago, same error occurred when I was damn-near done with a different flashpoint. The auto-targeting mechs that are close to sensor blips but still in the fog seems to be the only thing that I can think of that are the same, or could be the cause as I did see damage blips in the black when my poor Orion turned into a vegetable on me.

```
NullReferenceException: Object reference not set to an instance of an object
  at CBTBehaviorsEnhanced.HeatHelper.DesignMaskHeatMulti (BattleTech.Mech mech, System.Boolean isProjectedHeat) [0x00233] in <5ff2fe1a828c49df96fd57fb7aea4752>:0 
  at CBTBehaviorsEnhanced.HeatHelper.NormalizedAdjustedHeatSinkCapacity (BattleTech.Mech mech, System.Boolean isProjectedHeat, System.Boolean fractional) [0x00013] in <5ff2fe1a828c49df96fd57fb7aea4752>:0 
  at CBTBehaviorsEnhanced.HeatHelper.CalculateHeat (BattleTech.Mech mech, System.Int32 projectedHeat) [0x0003c] in <5ff2fe1a828c49df96fd57fb7aea4752>:0 
```

**Comment by IceRaptor** [comment]
Also reported in #ticket-3306 on RT server

Destroying allied mission relevant units (escort) via ground-targeting will lead to soft crash of the game:
- turn will not end by itself anymore after unit destruction
- no further events will be triggered
- workaround via "withdraw" doesn't work either (can be triggered, but after placement of withdraw area and audio you are stuck again)
- esc menu and "quit game" will crash the game
I was able to reproduce the bug twice, once with LRM Hydra ammo used, another time with LRM Swarm + SRM Deadfire ammo - so I guess the ammunition used is not the relevant facor
Personal note: Gameplay-wise this bug is especially problematic because in the current patch(es) extensive minefields are basically omni-present, oftentimes covering entire escort routes and eventually leading to escort vehicles/mechs being stuck very regularly (happened to me in both of the escort missions I was palying in Beta 1.2) without any reasonable chance to prevent or resolve the situation manually.

**Comment by IceRaptor** [comment]
Fixed in 0.4.5.0. Incorrectly re-used the designMask variable name.
